app/models/ext_management_system.rb

```diff
@@ -71,9 +71,9 @@ def self.supported_types_and_descriptions_hash
   def hostname_uniqueness_valid?
     return unless hostname_required?
     return unless hostname.present? # Presence is checked elsewhere
+    # check uniqueness per provider type
 
-    existing_hostnames = Endpoint.where.not(:resource_id => id).pluck(:hostname).compact.map(&:downcase)
-
+    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map(&:downcase)
     errors.add(:hostname, "has already been taken") if existing_hostnames.include?(hostname.downcase)
```

Comment:
How about 
```ruby
errors.add(:hostname, "has to be unique per provider type")
```

Also you have to wrap the message in `_()` to be translated. See https://github.com/durandom/manageiq/blob/master/app/models/dialog_tab.rb#L28